src/python/updaterecord-py.c

```diff
@@ -292,6 +292,7 @@ get_datetime(_UpdateRecordObject *self, void *member_offset)
             char *t;
             long long int epoch = strtoll(str, &t, 10);
             if(*t == '\0') {
+                g_free(dt);
                 return PyLong_FromLongLong(epoch);
             } else {
                 PyErr_SetString(CrErr_Exception, "Invalid date");
```

Comment:
Not related to the PR, but shouldn't you return NULL here? `dt` is empty here right, so the following code at the end of the function will not produce anything valid?

Also, while I'm commenting here, the Error could be more descriptive, as in saying what is the date it's trying to convert (where did it come from?) and actually include the broken date string too?

Tests for handling the error would be ideal... I didn't go looking where the string comes from.

Reply:
You are correct, it was the original behavior even before my epoch enhancement, but lets fix it. Also added a commit with one unit test for this invalid case.